btree.go

```diff
@@ -64,6 +64,51 @@ type Item interface {
 	Less(than Item) bool
 }
 
+// FreeList represents a free list of btree nodes. By default each
+// BTree has its own FreeList, but multiple BTrees can share the same
+// FreeList.
+type FreeList struct {
+	freelist []*node
+	grow     bool
+}
+
+func NewFreeList() *FreeList {
+	return newFreeList(true)
+}
+
+func newDefaultFreeList() *FreeList {
+	return newFreeList(false)
+}
+
+func newFreeList(grow bool) *FreeList {
```

Comment:
A valid concern: What if I introduced NewUnboundedFreeList() which returns a freelist where grow is true and change NewFreeList() so that it returns a freelist where grow is false. If I did this, the two statements you menetioned above would be functionally equivalent. For my app, I would use NewUnboundedFreeList() instead. Thoughts?